src/mono/browser/runtime/exports-binding.ts

```diff
@@ -114,6 +116,10 @@ export const mono_wasm_imports = [
     mono_wasm_get_first_week_of_year,
 ];
 
+if (NativeAOT) {
+    mono_wasm_imports.push(mono_wasm_bind_cs_function as any);
```

Comment:
We discussed yesterday why we need this: because of the `Interop.Runtime.BindCSFunction` call in the original fix that is/was a statically linked function.

Now the code calls `JavaScriptImports.BindCSFunction` that is a list of JSImports which are (presumably) dynamically resolved. So why is this still needed?

Reply:
I understand - the question is what code references `mono_wasm_bind_cs_function` statically now (it used to be line 286)?

Reply:
I'm not sure if I understand the question. But will try to answer it :)
We are never trimming any TS code based on emcc linker. It's separate `dotnet.runtime.js` module which is part of runtime pack. It is not part of the emcc compilation/linker on the customer dev machine. That's just `dotnet.native.js`

Reply:
Here's my question. First, we observe that the failure mode we saw when merging: https://github.com/dotnet/runtimelab/issues/2558#issuecomment-2074576866.
Q: what does this warning mean?
A: it means that emcc failed to resolve the `mono_wasm_bind_cs_function` symbol.
Q: why was that?
A: because it was removed from `exports-binding.ts`, which is the mechanism used to connect (via a rather complex scheme) TS code to Emscripten-produced code.
Q: but where was it referenced in the first place?
A: because the original fix called `Interop.Runtime.BindCSFunction` (on that line 286).

This change adds `mono_wasm_bind_cs_function` to `exports-binding.ts`, but _deletes_ the `Interop.Runtime.BindCSFunction` call. So, is it referenced somewhere else as well?

Reply:
Oh, great catch! The import is not needed now and I'll remove it. It was needed before I unified the call on C# side and didn't remove it afterwards

Reply:
so what is calling `mono_wasm_bind_cs_function` now ? something still has to do the marshaling on the JS side, right ?

Reply:
It's JSImport now

Reply:
so it will be on `INTERNAL` js object rather than C symbol.

Reply:
By the way - is this JSImport binding relative to `globalThis` or something else? In other words - will it work for the multiple-runtimes-on-the-same-page (aka native-library) scenario?

Reply:
JSImports are matched with objects passed `setModuleImports` with same shortcuts for matchin internal API. Nothing global involved

Reply:
there is `globalThis.getDotnetRuntime(0).INTERNAL` where runtime `runtimeId` is `0` if you are the first runtime installed in that page. 
You can get that id when you start the runtime. 

```
const runtime = await dotnet.create();
// runtime.runtimeId
// runtime.INTERNAL
```

You can have multiple runtimes on the same page, but you have to create them from separate ES6 modules.
```
const { dotnet: dotnet2 } = await import('./_framework/dotnet.js?instance=2-' + guid);
const runtime2 = await dotnet2.create();
```

The same ES6 module can't have multiple runtimes because emscripten/runtime instance state is not encapsulated enough.
For multi-threaded runtime it probably doesn't work, because there are `postMessage` with workers, but I'm not sure.